src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,112 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
+    private static final User VALID_USER = new User("username", "qwerty1", 20);
+    private static final User NEW_USER = new User("newUser", "qwerty1", 20);
+    private static final User INVALID_USER = new User("in_us", "qty1", 10);
+    private static final User NULL_USER = null;
+    private static final User UNDERAGE_USER = new User("underageUser", "qwerty1", 15);
+    private static final User INVALID_LOGIN_USER = new User("1n", "qwerty1", 20);
+    private static final User INVALID_PASSWORD_USER = new User("invalidPasswordUser", "in_p", 20);
+    private static final User NULL_LOGIN_USER = new User(null, "qwerty1", 15);
+    private static final User NULL_PASSWORD_USER = new User("nullPassword", null, 20);
+    private static final User NULL_AGE_USER = new User("nullAge", "qwety1", null);
+    private static StorageDaoImpl storage;
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    static void setUp() {
+        storage = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void userDataValidRegistration() {
```

Comment:
I think that it will be better if you rename tests, according to the task:

For this task use such convention: <methodUnderTest>_<state>_<expectedBehavior>. For example, if we are testing the method register with a null user’s age the test method name should be register_nullAge_notOk

Reply:
Agree